src/Data/Text/Encoding.hs

```diff
@@ -154,6 +154,10 @@ decodeLatin1 bs = withBS bs $ \fp len -> runST $ do
 foreign import ccall unsafe "_hs_text_is_ascii" c_is_ascii
     :: Ptr Word8 -> Ptr Word8 -> IO CSize
 
+isValidBS :: ByteString -> Bool
+isValidBS bs = withBS bs $ \fp len -> unsafeDupablePerformIO $
```

Comment:
This is an interesting point, thanks! It does not quite fit `text` public API, because it does not mention `Text`. Ideally this is something to expose from `bytestring` itself, but it's hardly justifiable to do so before `text` switches to UTF8. 

I've raised https://github.com/haskell/bytestring/issues/417 to track this. In the meantime let's keep it private.

Reply:
Sounds good, thanks @Bodigrim